core/src/main/scala-3/caliban/schema/SchemaDerivation.scala

```diff
@@ -44,16 +44,22 @@ trait CommonSchemaDerivation {
       case (_: EmptyTuple, _)                 => values.reverse
       case (_: (name *: names), _: (t *: ts)) =>
         recurseSum[R, P, names, ts] {
-          (
-            constValue[name].toString,
-            MagnoliaMacro.anns[t], {
-              if (Macros.isEnumField[P, t])
-                if (!Macros.implicitExists[Schema[R, t]]) derived[R, t]
-                else summonInline[Schema[R, t]]
-              else summonInline[Schema[R, t]]
-            }.asInstanceOf[Schema[R, Any]]
-          ) :: values
+          inline summonInline[Mirror.Of[t]] match {
```

Comment:
@ghostdogpr do you see any reason why we shouldn't do the same handling for `ArgBuilder`s?

Reply:
OK after thinking about it I added it to the ArgBuilder as well since this is what matches Scala 2's / Magnolias behaviour.

Reply:
Good call